airbyte-workers/src/main/java/io/airbyte/workers/temporal/check/connection/CheckConnectionActivityImpl.java

```diff
@@ -43,7 +45,8 @@ public CheckConnectionActivityImpl(final WorkerConfigs workerConfigs,
                                      final WorkerEnvironment workerEnvironment,
                                      final LogConfigs logConfigs,
                                      final JobPersistence jobPersistence,
-                                     final String airbyteVersion) {
+                                     final String airbyteVersion,
+                                     final FeatureFlags featureFlags) {
```

Comment:
Is it necessary to pass the FeatureFlags into each of these non-sync activities? From what I can tell, only the replication activity actually cares about the `USE_STREAM_CAPABLE_STATE` being set correctly.

I think passing this into all of these activities is fine if we are _not_ planning to remove this `featureFlags` object once we stop needing the USE_STREAM_CAPABLE_STATE flag. But if we are planning to remove this object completely when we stop using that flag, then it seems a bit overkill to pass it into every activity. What do you think?

Reply:
It was to avoid initializing it multiple times (only in workerApp). The env will be set in the container because of the way the integrationRunner is done but I can remove it from the code.